src/resolve.rs

```diff
@@ -253,15 +278,16 @@ impl std::error::Error for ResolveError {
 
 #[cfg(feature = "json")]
 mod json {
-    use super::{parse_index, Pointer, Resolve, ResolveError, ResolveMut};
+    use super::{parse_index, Error, Pointer, Resolve, ResolveMut};
     use serde_json::Value;
 
     impl Resolve for Value {
         type Value = Value;
-        type Error = ResolveError;
+        type Error = Error;
```

Comment:
Similar for `Assign`, but I wonder if the `Error` should just be `Error`, rather than an associated type. I know this theoretically allows external implementers to bring their own error type, but thinking of the design of libraries like `serde` and `std::io`, seems common to be a bit prescriptive with the type and just give a escape hatch for custom errors.

Not 100% on this one, but seeing how all the internal implementations share the same type makes me suspect this is overly abstract.